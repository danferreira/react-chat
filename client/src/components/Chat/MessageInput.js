import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTelegramPlane } from '@fortawesome/free-brands-svg-icons';

const MessageInputWrapper = styled.div`
    background: #fff;
    padding: 10px;
    order: 3;
    border: solid 1px #e8e8e8;
    position: relative;
`;

const Input = styled.input`
    width: 100%;
    resize: none;
    height: 37px;
    font-size: 1.2rem;
    color: #424242;
    padding: 10px 40px 10px 10px;
    border-radius: 10px;
    border: solid 1px #e8e8e8;
`;

const Button = styled.button`
    position: absolute;
    right: 25px;
    bottom: 12px;
    font-size: 30px;
    background-color: Transparent;
    background-repeat:no-repeat;
    border: none;
    cursor:pointer;
    overflow: hidden;
    outline:none;
`;

const propTypes = {
    onSend: PropTypes.func.isRequired
}

class MessageInput extends PureComponent {
    state = {
        message: ""
    };

    handleSubmit = (event) => {
        event.preventDefault();

        var value = this.state.message;

        if (!value.trim()) return;

        this.props.onSend(value);
        this.setState({ message: "" });
    }

    onChange = (event) => {
        this.setState({ message: event.target.value });
    }

    render() {
        return (
            <MessageInputWrapper>
                <form onSubmit={this.handleSubmit}>
                    <Input 
                        type="text" 
                        placeholder="Type a message" 
                        onChange={this.onChange} 
                        value={this.state.message} 
                        autoComplete="off" 
                        autoFocus />
                    <Button>
                        <FontAwesomeIcon icon={faTelegramPlane} />
                    </Button>
                </form>
            </MessageInputWrapper>
        );
    }
}

MessageInput.propTypes = propTypes;

export default MessageInput;