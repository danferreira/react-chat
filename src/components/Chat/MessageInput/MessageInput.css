.message-input {
    background: #fff;
    padding: 10px;
    order: 3;
    border: solid 1px #e8e8e8;
    position: relative;
}

.message-input input {
    width: 100%;
    resize: none;
    height: 37px;
    font-size: 1.2rem;
    color: #424242;
    padding: 10px 40px 10px 10px;
    border-radius: 10px;
    border: solid 1px #e8e8e8;
}

.message-input button {
    position: absolute;
    right: 25px;
    bottom: 12px;
    font-size: 30px;
    background-color: Transparent;
    background-repeat:no-repeat;
    border: none;
    cursor:pointer;
    overflow: hidden;
    outline:none;
}