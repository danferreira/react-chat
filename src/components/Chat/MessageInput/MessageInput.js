import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './MessageInput.css'

const propTypes = {
    onSend: PropTypes.func
}

class MessageInput extends Component {
    constructor(props) {
        super(props);
        this.state = {
            message: ""
        };
    }

    handleSubmit = (event) => {
        event.preventDefault();

        var value = this.state.message;
        
        if (!value.trim()) return;

        this.props.onSend(value);
        this.setState({ message: "" });
    }

    render() {
        return (
            <div className="message-input">
                <form onSubmit={this.handleSubmit}>
                    <input type="text" placeholder="Type a message" value={this.state.message} autoFocus autoComplete="off"></input>
                    <button type="submit">
                        <i className="fab fa-telegram-plane"></i>
                    </button>
                </form>
            </div>
        );
    }
}

MessageInput.propTypes = propTypes;

export default MessageInput;