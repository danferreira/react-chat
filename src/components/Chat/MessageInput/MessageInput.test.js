import React from 'react';
import { shallow, mount } from 'enzyme';

import MessageInput from './MessageInput';

describe('<MessageInput />', () => {

    it('should render properly', () => {
        const wrapper = shallow(<MessageInput />);
        expect(wrapper).toMatchSnapshot();
    })

    it('should call onChange function', () => {

        const wrapper = mount(<MessageInput />);
        const spyOnChange = jest.spyOn(wrapper.instance(), 'onChange');
        wrapper.instance().forceUpdate();

        wrapper.find('input').simulate('change');

        expect(spyOnChange).toHaveBeenCalled();

        spyOnChange.mockReset();
        spyOnChange.mockRestore();
    })

    it('should call onSend function', () => {
        const onSend = jest.fn();
        const value = 'example of some message';

        const wrapper = mount(<MessageInput onSend={onSend} />);

        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.find('button').simulate('submit');

        expect(onSend).toBeCalledWith(value);

        onSend.mockReset();
    })
});