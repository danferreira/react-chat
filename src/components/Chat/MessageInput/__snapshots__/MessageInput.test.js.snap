// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageInput /> should call function 1`] = `
<MessageInput
  onSend={[Function]}
>
  <div
    className="message-input"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        autoComplete="off"
        autoFocus={true}
        onChange={[Function]}
        placeholder="Type a message"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        <i
          className="fab fa-telegram-plane"
        />
      </button>
    </form>
  </div>
</MessageInput>
`;

exports[`<MessageInput /> should render properly 1`] = `
<div
  className="message-input"
>
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      autoFocus={true}
      onChange={[Function]}
      placeholder="Type a message"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      <i
        className="fab fa-telegram-plane"
      />
    </button>
  </form>
</div>
`;
